### ISSUE TITLE
Detect and warn or error about assigning different type aliases using the same basetype.

### ISSUE THREAD

**Issue by thargol1** [dotnet/roslyn#68186]
Some example code to explain my question

```csharp
using EmployeeId = Guid;
using StudentId = Guid;

EmployeedId e = Guid.NewGuid();
StudentId s = e; // <-- this should raise a warning or an error
Guid g = e; // <-- this is ok
StudentId s2 = g; // <-- this is ok for the compiler, but a programmer's mistake
```
Perhaps the syntax should be augmented to enable this or it could be a compiler switch.
```csharp
using locked EmployeeId = Guid;  // <-- the locked keyword make this type only assignable by it self or it's basetype.
using locked StudentId = Guid;
```

For example: most database-id's or either `Guid` or `int`. This feature could prevent accidentally assigning non related id's to each other, by giving each model it's own id-type.

**Comment by CyrusNajmabadi** [comment]
You could write an analyzer to do this.

**Comment by jcouv** [comment]
It's possible that explicit extensions (feature still being designed) could help solve this problem. Alternatively, an analyzer could also be a possible solution.
But as things stand, the behavior is by-design. If you'd like to propose a language change, consider starting a discussion on the csharplang repo.
